internal/artifacts/module_arch_test.go

```diff
@@ -89,97 +90,142 @@ builders:
 				Ω(os.Remove(getTestPath("mtaModelsBuild", "ui5app", "test2_copy.txt"))).Should(Succeed())
 				Ω(os.Remove(getTestPath("mtaModelsBuild", "ui5app2", "test2_copy.txt"))).Should(Succeed())
 			})
-			It("required module m2 has ready artifact 'test2.txt' and creates a new one 'test2_copy.txt'", func() {
-				Ω(ExecuteSoloBuild(getTestPath("mtaModelsBuild"), getResultPath(), nil, []string{"m1", "m3"}, true, os.Getwd)).Should(Succeed())
+
+			It("required module m2 has ready artifact 'test2.txt' and creates a new one 'test2_copy.txt', mtad.yaml generated", func() {
+				Ω(ExecuteSoloBuild(getTestPath("mtaModelsBuild"), getResultPath(), nil, []string{"m1", "m3"}, true, true, "cf", os.Getwd, yaml.Marshal)).Should(Succeed())
 				Ω(getTestPath("result", "data.zip")).Should(BeAnExistingFile())
 				Ω(getTestPath("result", "m3.zip")).Should(BeAnExistingFile())
 				validateArchiveContents([]string{"test.txt", "test2.txt", "test2_copy.txt"}, getTestPath("result", "data.zip"))
+				mtadContent, err := ioutil.ReadFile(getTestPath("result", "mtad.yaml"))
+				Ω(err).Should(Succeed())
+				mtadObj, err := mta.Unmarshal(mtadContent)
+				Ω(err).Should(Succeed())
+				for _, module := range mtadObj.Modules {
+					if module.Name == "m1" {
+						Ω(module.Path).Should(Equal(getTestPath("result", "data.zip")))
+					} else if module.Name == "m3" {
+						Ω(module.Path).Should(Equal(getTestPath("result", "m3.zip")))
+					}
+				}
+			})
+
+			Describe("mtad.yaml generation fails on corrupted platform configuration", func() {
+
+				var platformConfig []byte
+
+				BeforeEach(func() {
+					platformConfig = platform.PlatformConfig
+					platform.PlatformConfig = []byte("corrupted")
+				})
+
+				AfterEach(func() {
+					platform.PlatformConfig = platformConfig
+				})
+
+				It("failure", func() {
```

Comment:
`failure` is not a descriptive test name - please rename to `mtad.yaml generation fails` (along with the change in the `Describe` I wrote above)

Reply:
changed